modules/apps/client-extension/client-extension-web/src/main/resources/META-INF/resources/js/components/FileEntryPicker.tsx

```diff
@@ -0,0 +1,351 @@
+		fileName: string,
+		isValidatingJSON: boolean
+	) => {
+		if (isValidatingJSON) {
+			return Liferay.Language.get('validating-json');
+		}
+
+		if (fileName) {
+			return fileName;
+		}
+
+		return Liferay.Language.get('no-json-selected');
+	};
+
+	const onInputChange = async (event: any) => {
+		const target = event.target as HTMLInputElement;
+
+		const filePath = target.value;
+
+		const fileName = filePath.replace(/^.*[\\]/, '');
+
+		setFileName(fileName);
+
+		if (!fileName.endsWith('.json')) {
+			setFeedback({
+				displayType: 'danger',
+				message: Liferay.Language.get(
+					'the-format-is-not-valid-please-upload-a-valid-frontend-token-definition-json-file'
+				),
+			});
+
+			setFrontendTokenDefinitionRefValue(MALFORMED_JSON);
+
+			return;
+		}
+
+		setIsValidatingJSON(true);
+
+		if (!alertRole) {
+			setAlertRole('alert');
+		}
+
+		setFeedback({
+			displayType: 'info',
+			message: Liferay.Language.get(
+				'the-frontend-token-definition-json-file-is-being-uploaded-and-validated'
+			),
+		});
+
+		readInputFile(fileName, target);
+	};
+
+	function readInputFile(fileName: string, input: HTMLInputElement) {
+		if (input.files === null) {
+			return;
+		}
+
+		let file = input.files[0];
+
+		const fileReader = new FileReader();
+
+		fileReader.onload = async function (event) {
+			let frontendTokenDefinitionString = event.target?.result as string;
+
+			if (frontendTokenDefinitionString === '') {
+				file = new File(['{}'], fileName, {
+					type: 'application/json',
+				});
+				frontendTokenDefinitionString = '{}';
+			}
+
+			if (frontendTokenDefinitionRef.current) {
+				frontendTokenDefinitionRef.current.value = frontendTokenDefinitionString;
+			}
+
+			await handleUpload(file);
+
+			setIsValidatingJSON(false);
+		};
+
+		fileReader.readAsText(file);
+	}
+
+	const handlePost = async (body: any, endpoint: string) => {
+		try {
+			const response = await Liferay.Util.fetch(
+				'/o/com-liferay-frontend-token-definition-impl' + endpoint,
+				{
+					body,
+					method: 'POST',
+				}
+			);
+
+			const data = await response.json();
+
+			let displayType: DisplayType = 'success';
+
+			if (!response.ok) {
+				displayType = 'danger';
+
+				setFrontendTokenDefinitionRefValue(MALFORMED_JSON);
+			}
+
+			setFeedback({
+				displayType,
+				message: data.message,
+			});
+		}
+		catch (error) {
+			clearSelection();
+
+			setFeedback({
+				displayType: 'danger',
+				message: Liferay.Language.get(
+					'an-error-has-occurred-while-connecting-to-our-servers'
+				),
+			});
+		}
+	};
+
+	async function handleUpload(
+		file: File,
+		focusSelectFileButton: boolean = true
+	) {
+		const formData = new FormData();
+
+		formData.append('file', file);
+
+		await handlePost(formData, '/validate-file');
+
+		if (focusSelectFileButton) {
+			selectFileButtonRef.current?.focus();
+		}
+	}
+
+	const getClientExtensionFrontendTokenDefinitionInfo = () => {
+		handlePost(
+			Liferay.Util.objectToFormData({
```

Comment:
objectToFormData should be imported from frontend-js-web. Also can we use this in the other instance where we have manually created formData?